catalog/ui/search-ui/standard/src/main/java/org/codice/ddf/ui/searchui/standard/properties/ConfigurationStore.java

```diff
@@ -265,22 +259,75 @@ public void setTerrainProvider(String terrainProvider) {
         setProxyForTerrain(terrainProvider);
     }
 
-    private void setProxiesForImagery(List<String> imageryProviders) {
-        if (imageryEndpoints.size() > 0) {
-            for (String endpoint : imageryEndpoints) {
-                try {
-                    httpProxy.stop(endpoint);
-                } catch (Exception e) {
-                    LOGGER.error("Unable to stop proxy endpoint.", e);
+    private void setProxiesForImagery(List<String> proxiesForImagery) {
+        List<Map<String, Object>> newImageryProviders = proxiesForImagery.stream()
+                .map(this::getProviderConfigMap)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        List<Map<String, Object>> imageryProvidersToStop = new ArrayList<>();
+        List<Map<String, Object>> imageryProvidersToStart = new ArrayList<>();
+
+        findDifferences(imageryProviderMaps, newImageryProviders, imageryProvidersToStart);
+        findDifferences(newImageryProviders, imageryProviderMaps, imageryProvidersToStop);
+
+        List<String> proxiesToStop = imageryProvidersToStop.stream()
+                .map(provider -> urlToProxyMap.get(provider.get(URL)))
+                .collect(Collectors.toList());
+
+        stopImageryEndpoints(proxiesToStop);
+        for (Map<String, Object> providerToStop : imageryProvidersToStop) {
+            urlToProxyMap.remove(providerToStop.get(URL));
+        }
+        startImageryEndpoints(imageryProvidersToStart);
+        proxiedImageryProviders.clear();
+        for (Map<String, Object> newImageryProvider : newImageryProviders) {
+            HashMap<String, Object> map = new HashMap<>(newImageryProvider);
+            map.put(URL, SERVLET_PATH + "/" + urlToProxyMap.get(newImageryProvider.get(URL)));
+            proxiedImageryProviders.add(map);
+        }
+        imageryProviderMaps = newImageryProviders;
+    }
+
+    private void findDifferences(List<Map<String, Object>> innerList,
```

Comment:
Can still simplify this method with one of the options mentioned on the previous commit.